### PR TITLE
[YUNIKORN-1369] Shim: Read configuration from configmaps and deprecate legacy configs

### PR DESCRIPTION
**Pull request by craigcondit**
### What is this PR for?
Read configuration from new ConfigMap entries in addition to legacy values, and deprecate old configuration sources.

[WIP] Based on and depends on from YUNIKORN-1365.

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-1369

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/client/kubeclient.go [hunk 1]
```diff
@@ -73,6 +68,29 @@ func newSchedulerKubeClient(kc string) SchedulerKubeClient {
 	}
 }
 
+func createRestConfig(kc string) *rest.Config {
+	// attempt to use in-cluster config
+	config, err := rest.InClusterConfig()
+	if err != nil && err != rest.ErrNotInCluster {
+		// this is called during early initialization, so abort
+		panic(err)
```
Comment (pbacsko) [inline pkg/client/kubeclient.go:76]: Nit: how about `log.Logger.Panic()` or `log.Logger.Fatal()`?

File: pkg/common/constants/constants.go [hunk 2]
```diff
@@ -56,7 +57,7 @@ const DaemonSetType = "DaemonSet"
 const AppManagerHandlerName = "yunikorn-app"
 
 // Gang scheduling
-const PlaceholderContainerImage = "k8s.gcr.io/pause"
+const PlaceholderContainerImage = "registry.k8s.io/pause:3.7"
```
Comment (pbacsko) [inline pkg/common/constants/constants.go:60]: This image is hard-coded at various places, like in the `Dockerfile` or `values.yaml`, do those need to be changed?

File: pkg/client/kubeclient.go (hunk 1 above)
Comment (craigcondit) [reply pkg/client/kubeclient.go:76]: Circular dependencies. This is called before logging is safe to call. Panic will log the error and abort, so we still get it. It's also pretty unlikely to fail in real life.

File: pkg/common/constants/constants.go (hunk 2 above)
Comment (craigcondit) [reply pkg/common/constants/constants.go:60]: The other places will be going away once this PR is complete, as we are transitioning to having defaults be in code only.

File: pkg/cache/context.go [hunk 3]
```diff
@@ -290,33 +295,79 @@ func (ctx *Context) filterConfigMaps(obj interface{}) bool {
...
-	ctx.triggerReloadConfig(nil)
-}
+	var configmap *v1.ConfigMap = nil
+	switch t := obj.(type) {
+	case *v1.ConfigMap:
+		configmap = t
+	case cache.DeletedFinalStateUnknown:
+		configmap = utils.Convert2ConfigMap(obj)
+	default:
+		log.Logger().Debug("unable to convert to configmap")
```
[the remark this answers is not in the document]
Comment (craigcondit) [reply pkg/cache/context.go:336]: Fair point. I'll update in the next version.

**Review by pbacsko — APPROVED** [review]
+1 LGTM (pending small change related to logging)

**Comment by craigcondit** [comment]
Merged to master, updated Debug() to Warn() as requested.

File: pkg/client/kubeclient.go (hunk 1 above)
Comment (craigcondit) [reply pkg/client/kubeclient.go:76]: Hmm, noticed after commit that we can in fact do this. Will update in the next PR in this series.

**Pull request closed without merging**